library/src/scala/quoted/Quotes.scala

```diff
@@ -3840,6 +3840,16 @@ trait Quotes { self: runtime.QuoteUnpickler & runtime.QuoteMatching =>
       // TODO: add flags and privateWithin
       @experimental def newClass(parent: Symbol, name: String, parents: List[TypeRepr], decls: Symbol => List[Symbol], selfType: Option[TypeRepr]): Symbol
 
+      /*
+       * @param paramNames constructor parameter names.
+       * @param paramTypes constructor parameter types.
+       * @param flags extra flags with which the class symbol should be constructed.
+       * @param privateWithin the symbol within which this new method symbol should be private. May be noSymbol.
+       *
+       * Parameters can be obtained via classSymbol.memberField
+       */
+      @experimental def newClass(parent: Symbol, name: String, parents: List[TypeRepr], decls: Symbol => List[Symbol], selfType: Option[TypeRepr], paramNames: List[String], paramTypes: List[TypeRepr], flags: Flags, privateWithin: Symbol): Symbol
```

Comment:
would it be possible to reformulate such that `parents` is also a function (to support param refs?)

Reply:
Do you mean that it should be a `Symbol => List[TypeRepr]` type? I have some trouble visualizing where we would need that classSymbol to define the parent TypeRepr (`class Cls extends Cls.InnerCls`?, is that legal?). Could you give me an example of a tree like that?

Reply:
Just saw `class Cls extends Cls.InnerCls` in the wild but in that case, Cls also had a companion object which contained InnerCls - not sure if that is possible to define right now but I'll have to look into the newModule method and see if it also needs updates

Reply:
I mean something like `class anon(x: Int) extends Box[x.type]`